azure-pipelines.yml

```diff
@@ -0,0 +1,71 @@
+# Node.js with Angular
+# Build a Node.js project that uses Angular.
+# Add steps that analyze code, save build artifacts, deploy, and more:
+# https://docs.microsoft.com/azure/devops/pipelines/languages/javascript
+variables:
+  appDir: 'fakebook/'
+
+trigger:
+  branches:
+   include: 
+     - main
+  paths:
+    include:
+      - az-pipeline-angular.yml
+
+pool:
+  vmImage: 'ubuntu-latest'
+
+stages:
+  - stage: build
+    jobs:
+    - job: build
+      variables:
+        imageName: 'fakebookng'
+      
+      steps:
+      - task: Docker@2
+        displayName: Build Docker Image
+        inputs:
+          repository: $(imageName)
+          command: 'build'
+          Dockerfile: '$(appdir)Dockerfile'
+  
+  - stage: SonarCloud
+    jobs:
+    - job: build
+      steps:
+      - task: NodeTool@0
+        inputs:
+          versionSpec: '10.x'
+        displayName: 'Install Node.js'
+
+      - script: |
+          npm install -g @angular/cli
+          npm install
+          ng build --prod
+        continueOnError: True
+        displayName: 'npm install and build'
+        workingDirectory: $(appDir)
```

Comment:
this is a slow way of doing things, instead run `npm ci` and then `npm run build -- --prod` or `npx ng build --prod`
and do it in two separate `script` steps; azure pipelines doesn't propagate errors except from the last command in a multiline script.

Reply:
Separated all out. also use npm ci and npx ng build --prod